install_files/ansible-base/roles/ossec-server/templates/send_encrypted_alarm.sh

```diff
@@ -1,5 +1,49 @@
 #!/bin/bash
+# Handler script to encrypt OSSEC alerts prior to mailing.
+# Called via the `.procmailrc` for user `ossec`.
 
-/usr/bin/formail -I "" | \
-    /usr/bin/gpg --homedir /var/ossec/.gnupg --trust-model always -ear "{{ ossec_gpg_fpr }}" | \
-    /usr/bin/mail -s "$(echo $SUBJECT | sed -r 's/([0-9]{1,3}\.){3}[0-9]{1,3}\s?//g' )" {{ ossec_alert_email }}
+# Intentionally NOT setting `set -e`, since we'll do our own error handling.
+# Rather than exit immediately on errors, we should gracefully inform
+# the Admin that there was a problem encrypting a message.
+#set -e
+set -o pipefail
+
+
+# Read alert message content from STDIN, as passed by OSSEC. Store the message
+# in a var so we can iteratively process it, halting execution before sending
+# if the GPG encryption command failed. If we simply chain a bunch of pipes
+# together, then the email will be sent with an empty body, should the GPG
+# encryption fail.
+ossec_alert_text="$(< /dev/stdin)"
+
+# Primary "send email to Admin" functionality.
+function send_encrypted_alert() {
+
+    local encrypted_alert_text
+    # Try to encrypt the alert message. We'll inspect the exit status of the
+    # pipeline to decide whether to send the alert text, or the default
+    # failure message.
+    encrypted_alert_text="$(printf "${ossec_alert_text}" | \
+        /usr/bin/formail -I '' | \
+        /usr/bin/gpg --homedir /var/ossec/.gnupg --trust-model always -ear '{{ ossec_gpg_fpr }}')"
```

Comment:
I think you should add `--no-batch` here as it is "highly recommended" in the GPG manual. `--with-colons`, also recommended for unattended mode is irrelevant here, but you could specify the last recommended option for unattended use by adding the flag `--status-fd 2` and then putting `2> /dev/null` after the GPG command. Also, to make it explicit the input is going to come from stdin you'd end the GPG args with `-`. Of course this works just the way it is, but my preference is to be explicit and follow all recommendations.

>        --no-batch
              Use batch mode.  Never ask, do not allow  interactive  commands.   --no-batch  disables
              this option.  Note that even with a filename given on the command line, gpg might still
              need to read from STDIN (in particular if gpg figures that the input is a detached sig‐
              nature  and no data file has been specified).  Thus if you do not want to feed data via
              STDIN, you should connect STDIN to g‘/dev/null’.

              It is highly recommended to use this option along  with  the  options  --status-fd  and
              --with-colons for any unattended of gpg.

>        --status-fd n
              Write  special  status  strings  to the file descriptor n.  See the file DETAILS in the
              documentation for a listing of them.


>       --with-colons
              Print  key  listings delimited by colons. Note that the output will be encoded in UTF-8
              regardless of any --display-charset setting. This format is useful when GnuPG is called
              from  scripts  and  other  programs as it is easily machine parsed. The details of this
              format are documented in the file ‘doc/DETAILS’, which is included in the GnuPG  source
              distribution.